apps/api/src/app/admin/admin.controller.ts

```diff
@@ -313,6 +314,42 @@ export class AdminController {
     return this.adminService.getMarketDataBySymbol({ dataSource, symbol });
   }
 
+  @UseGuards(AuthGuard('jwt'))
+  @Post('market-data/:dataSource/:symbol')
+  public async updateMarketData(
+    @Param('dataSource') dataSourceParam: DataSource,
+    @Param('symbol') symbolParam: string,
+    @Body() data: UpdateBulkMarketDataDto
+  ) {
+    if (
+      !hasPermission(
+        this.request.user.permissions,
+        permissions.accessAdminControl
+      )
+    ) {
+      throw new HttpException(
+        getReasonPhrase(StatusCodes.FORBIDDEN),
+        StatusCodes.FORBIDDEN
+      );
+    }
+    const dataBulkUpdate: Prisma.MarketDataUpdateInput[] = [];
+    data.marketData.forEach((entry) => {
```

Comment:
You can probably combine it like this:
```
const dataBulkUpdate = data.marketData.forEach(({ date, marketPrice }) => {
  dataBulkUpdate.push({
    date,
    marketPrice,
    dataSource: dataSourceParam,
    state: 'CLOSE',
    symbol: symbolParam
  });
});
```

Reply:
close enough

`  
const dataBulkUpdate: Prisma.MarketDataUpdateInput[] = data.marketData.map(
      (entry) => ({
        dataSource: dataSource,
        symbol: symbol,
        date: entry.date,
        marketPrice: entry.marketPrice,
        state: 'CLOSE'
      })
    );
`